web/reports/views.py

```diff
@@ -26,13 +26,20 @@
 
 
 class BaseReportView(LoginRequiredMixin, PermissionRequiredMixin):
-    permission_required = [Perms.page.view_reports]
+    permission_required = [Perms.sys.access_reports]
+
+    def has_permission(self) -> bool:
+        has_permission = super().has_permission()
+        if not has_permission:
+            return has_permission
+        return can_user_view_report(self.get_report(), self.request.user)
```

Comment:
would just turn this into:

```suggestion
        return can_user_view_report(self.get_report(), self.request.user) and super().has_permission()
```